core/src/commonMain/kotlin/pro/respawn/flowmvi/StoreImpl.kt

```diff
@@ -77,25 +76,27 @@ internal class StoreImpl<S : MVIState, I : MVIIntent, A : MVIAction>(
         onAction = { action -> onAction(action)?.let { _actions.action(it) } },
         onTransformState = { transform -> this@StoreImpl.updateState { onState(this, transform()) ?: this } },
         onStop = { e -> close().also { plugin.onStop?.invoke(this, e) } },
-        onStart = { lifecycle ->
+        onStart = pipeline@{ lifecycle ->
             beginStartup(lifecycle)
-            launch intents@{
-                coroutineScope {
-                    // run onStart plugins first to not let subscribers appear before the store is started fully
-                    if (plugin.onStart != null) catch { onStart() }
-                    if (plugin.onSubscribe != null || plugin.onUnsubscribe != null) launch {
-                        observeSubscribers(
-                            onSubscribe = { catch { onSubscribe(it) } },
-                            onUnsubscribe = { catch { onUnsubscribe(it) } }
-                        )
-                    }
-                    if (plugin.onIntent != null) intents.awaitIntents {
-                        catch {
-                            val result = onIntent(it)
-                            if (result != null && config.debuggable) throw UnhandledIntentException(result)
-                        }
+            val startup = launch {
+                if (plugin.onStart != null) catch { onStart() }
+                lifecycle.completeStartup()
+            }
```

Comment:
Remember that catch { } is a custom function I defined on RecoverModule for proper handling of recovery actions. Inside Store implementations, catch { } should be used instead of the regular try-catch. Thus, this is not a valid comment.